code/game/machinery/dance_machine.dm

```diff
@@ -420,59 +424,13 @@
 				initial_matrix.Translate(-3,0)
 				animate(M, transform = initial_matrix, time = 1, loop = 0)
 		sleep (1)
-	animate(M, transform = null, time = 1, loop = 0)
-
-/obj/machinery/disco/proc/dance5helper(var/mob/living/carbon/M)
-	if (M)
-		animate(M, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
-	sleep (70)
-	if (M)
-		animate(M, transform = null, time = 1, loop = 0)
-
-/mob/living/carbon/proc/dancey() // Dance5 except independent of the machine if admins want to meme it up
-	INVOKE_ASYNC(src, .proc/danceyhelper)
-	var/matrix/initial_matrix = matrix(transform)
-	for (var/i in 1 to 60)
-		if (!src)
-			return
-		if (i<31)
-			initial_matrix = matrix(transform)
-			initial_matrix.Translate(0,1)
-			transform = initial_matrix
-			animate(src, transform, time = 1, loop = 0)
-		if (i>30)
-			initial_matrix = matrix(transform)
-			initial_matrix.Translate(0,-1)
-			transform = initial_matrix
-			animate(src, transform, time = 1, loop = 0)
-		setDir(turn(src.dir, 90))
-		switch (dir)
-			if (NORTH)
-				initial_matrix = matrix(transform)
-				initial_matrix.Translate(0,3)
-				animate(src, transform, time = 1, loop = 0)
-			if (SOUTH)
-				initial_matrix = matrix(transform)
-				initial_matrix.Translate(0,-3)
-				animate(src, transform, time = 1, loop = 0)
-			if (EAST)
-				initial_matrix = matrix(transform)
-				initial_matrix.Translate(3,0)
-				animate(src, transform, time = 1, loop = 0)
-			if (WEST)
-				initial_matrix = matrix(transform)
-				initial_matrix.Translate(-3,0)
-				animate(src, transform, time = 1, loop = 0)
-		sleep (1)
-	animate(src, transform = null, time = 1, loop = 0)
+	M.lying_fix()
+
 
-/mob/living/carbon/proc/danceyhelper()
-	if (src)
-		animate(src, transform = matrix(180, MATRIX_ROTATE), time = 1, loop = 0)
-	sleep (70)
-	if (src)
-		animate(src, transform = null, time = 1, loop = 0)
 
+/mob/living/proc/lying_fix()
+	animate(src, transform = null, time = 1, loop = 0)
+	lying_prev = 0
```

Comment:
Incorrect, lying_prev represents the number of degrees that the sprite is turned, between 0 and 360.